include/boost/math/tools/polynomial_gcd.hpp

```diff
@@ -19,38 +19,19 @@
 #ifndef BOOST_MATH_STANDALONE
 #include <boost/integer/common_factor_rt.hpp>
 
-// std::gcd was introduced in C++17
-#elif (__cplusplus > 201700L || _MSVC_LANG > 201700L)
-
-#define BOOST_MATH_CXX17_NUMERIC
+#else
 #include <numeric>
 #include <utility>
 #include <iterator>
 #include <type_traits>
 #include <boost/math/tools/assert.hpp>
 
-namespace boost::integer {
-
-template <typename Iter, typename T = typename std::iterator_traits<Iter>::value_type>
-std::pair<T, Iter> gcd_range(Iter first, Iter last) noexcept(std::is_arithmetic_v<T>)
-{
-    using std::gcd;
-    BOOST_MATH_ASSERT(first != last);
-
-    T d = *first;
-    ++first;
-    while (d != T(1) && first != last)
-    {
-        d = gcd(d, *first);
-        ++first;
-    }
-    return std::make_pair(d, first);
-}
+namespace boost { namespace integer {
 
 namespace gcd_detail {
 
 template <typename EuclideanDomain>
-inline EuclideanDomain Euclid_gcd(EuclideanDomain a, EuclideanDomain b) noexcept(std::is_arithmetic_v<EuclideanDomain>)
+inline EuclideanDomain Euclid_gcd(EuclideanDomain a, EuclideanDomain b) noexcept(std::is_arithmetic<EuclideanDomain>::value)
```

Comment:
Can this not defer to std::gcd when available?  Euclid's method is very straightforward, but other methods (such as binary gcd) are usually faster.